2018/002-update-channels.md

```diff
@@ -0,0 +1,207 @@
+is displayed. A system with access to the web can fetch the release notes and
+display them in the UI directly. In both scenarios, an admin can evaluate the
+upcoming changes beforehand.
+
+The UI of velum needs to be adapted to enable a switch to a new channel.
+
+## Detailed RFC
+
+We can separate the updates into different channels. There are 2 types of channels:
+
+* update channel: normal maintenance updates, which are safe and non-intrusive
+* new version channel: version updates which can contain intrusive updates
+
+This RFC proposes:
+
+* a front-end change in the velum UI to enable the admin to switch to a new version
+channel
+* a back-end change in salt to perform the channel switch on all cluster nodes
+
+### Proposed change (Detailed)
+
+#### Tools/Services
+
+We need a set of tools/services to indicate, whenever a maintenance update or a new version
+update is ready to be installed. They should run on each worker node to be more independent
+and decoupled from velum. If velum would be the initiator of the update check, it might lead
+to inconsistent and long execution runs in larger clusters.
+
+##### "check for updates" service on each worker
+
+The current channel will offer maintenance updates, which need to be announced.
+
+The service:
+
+* runs once per day e.g.
+* is generally decoupled from velum, runs independently, and just reports.
+* reports availability of maintenance updates to velum via grains.
+
+##### "new version" checker service on each worker
+
+The next channel will offer possibly disruptive updates, which can be either features or
+version updates of certain components of the whole stack, which need to be announced.
+
+The service:
+
+* runs once per week or longer timeframe.
+* is generally decoupled from velum, runs independently, just reports.
+* checks for new versions (migration targets):
+  - query SCC API if there is a new compatible version available for this CaaSP version (SUSEConnect library)
+  - reports availability of new versions to velum via grains.
+  Furthermore if the system is registered against SMT/RMT:
+  - query RMT/SMT API if there is a new compatible version available for this CaaSP version
+  - reports if an admin needs to mirror the new version channel completely first (out of sync mirror)
+* get the link to the release notes
+
+also see section "Systems Registered against RMT/SMT"
+
+#### User Interface
+
+The user interface of velum needs to be adapted minimally.
+
+* [indirect] display of update content through [a link to the] release notes
+* display of update history via velum (optional)
+* button to trigger the maintenance updates (what we currently have)
+* button to switch to the next channel in velum
+
+#### Salt back-end
+
+We will continue to use `transactional-update` to trigger the updates and migrations.
+Salt will be used to call `transactional-update [dup|migrate]` as it is done now already.
```

Comment:
Currently, transactional-update is called by a systemd timer, not by salt. This is something we need to change.